kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveKeyValueParams.java

```diff
@@ -46,7 +46,16 @@ public static ResponsiveKeyValueParams fact(final String name) {
   }
 
   public ResponsiveKeyValueParams withTimeToLive(final Duration timeToLive) {
-    this.timeToLive = timeToLive;
+    return withTtlProvider(TtlProvider.withDefault(timeToLive));
+  }
+
+  public ResponsiveKeyValueParams withTtlProvider(final TtlProvider<?, ?> ttlProvider) {
+    if (ttlProvider.hasConstantTtl() && !ttlProvider.defaultTtl().isFinite()) {
```

Comment:
```suggestion
    if (ttlProvider.hasDefaultOnly() && !ttlProvider.defaultTtl().isFinite()) {
```
nit: `hasConstantTtl` trips me up a little bit, might as well just directly say what it is

Reply:
I guess to me, "has constant" is more "what it is" than "default only" hahaha. But I can also see it your way. I'll change it